Cargo.toml

```diff
@@ -86,15 +86,39 @@ proptest-derive = { version = "0.7", default-features = false }
 serde = { version = "1.0", default-features = false, features = ["alloc", "derive", "rc"] }
 serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
 serde-untagged = {version = "0.1" }
-sha2 = { version = "0.10", default-features = false }
 smallvec = { version = "1.15", default-features = false, features = ["union", "const_generics", "const_new"] }
 thiserror = { version = "2.0", default-features = false }
 tokio = { version = "1.48", default-features = false }
 tracing = { version = "0.1", default-features = false, features = ["attributes"] }
-winter-air = { version = "0.13", default-features = false }
-winter-fri = "0.13"
-winter-math = { version = "0.13", default-features = false }
-winter-prover = { version = "0.13", default-features = false }
-winter-rand-utils = "0.13"
-winter-utils = { version = "0.13", default-features = false }
-winter-verifier = { version = "0.13", default-features = false }
+rand = { version = "0.9", default-features = false }
+sha2 = { version = "0.10", default-features = false }
+
+# Upstream Plonky3 dependencies
+p3-air           = { version = "0.4", default-features = false }
+p3-blake3        = { version = "0.4", default-features = false }
+p3-challenger    = { version = "0.4", default-features = false }
+p3-commit        = { version = "0.4", default-features = false }
+p3-dft           = { version = "0.4", default-features = false }
+p3-field         = { version = "0.4", default-features = false }
+p3-interpolation = { version = "0.4", default-features = false }
+p3-keccak        = { version = "0.4", default-features = false }
+p3-matrix        = { version = "0.4", default-features = false }
+p3-maybe-rayon   = { version = "0.4", default-features = false }
+p3-merkle-tree   = { version = "0.4", default-features = false }
+p3-mds           = { version = "0.4", default-features = false }
+p3-monty-31      = { version = "0.4", default-features = false }
+p3-poseidon2     = { version = "0.4", default-features = false }
+p3-symmetric     = { version = "0.4", default-features = false }
+p3-uni-stark     = { version = "0.4", default-features = false }
+p3-util          = { version = "0.4", default-features = false }
```

Comment:
Without having looked too deeply into the code: do we need all of these dependencies? For example, what's the purpose of `p3-monty-31` or `p3-uni-stark` dependency?

Also, in the longer run, we could consider re-exporting some of the crates through our adaptations. For example, we could could probably get w/e we need from `p3-air` via ` pr-miden-air`.

Reply:
Agreed, the most immediate one is `p3-monty-31` which I would be very surprised if it is a real dependence.
I think, originally, these were dev leftovers or the result of a copy-paste.